### PR TITLE
Fix decorator abstract methods validation to match CDI specification

### LINKED ISSUE #51196: ArC: imprecise check for abstract methods in decorators
### Describe the bug

The check for `abstract` methods declared in decorators, as present in `Decorators.createDecorator()`, is imprecise. We check that decorators do not declare _any_ `abstract` methods, but the spec says that decorators may not declare `abstract` methods that do not belong to a decorated type. Also, we only check methods declared on the decorator class itself, not inherited methods. (We can probably ignore `abstract` methods inherited from interfaces, because those are decorated types, but a decorator can inherit `abstract` methods from a superclass, and those do not necessarily have to be present on decorated types.)

That doesn't cause any TCK failures, because decorators are not present in CDI Lite (plus I'm not sure this is actually tested by the TCK), and so far there hasn't been any issues filed about this, so this is purely of a theoretical concern (I only stumbled upon this by writing test cases for various corner cases). Still, it is a deviation from the specification, so I believe we should fix it. It's a lower priority kind of thing, though.

### Expected behavior

Only fail when an `abstract` method that doesn't belong to a decorated type is explicitly declared on a decorator or inherited (and not implemented!) from a supertype.

### Actual behavior

Fail whenever an `abstract` method is explicitly declared on a decorator; `abstract` methods declared on supertypes are ignored.

### How to Reproduce?

N/A

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by fabriciolima**
The current check for abstract methods in decorators was too restrictive. According to the CDI specification, decorators may declare abstract methods that belong to decorated types. The check should only reject abstract methods that do not belong to any decorated type.

Changes:
- Modified Decorators.createDecorator() to check if abstract methods belong to decorated types before rejecting them
- Added belongsToDecoratedType() method to verify if a method exists in any decorated type
- Added methodParametersMatch() helper to compare method signatures
- Now checks abstract methods from decorator class and inherited from superclasses (as mentioned in the TODO comment)

This fixes the issue where decorators with abstract methods that belong to the decorated interface were incorrectly rejected.

- Fixes: #51196

### CODE REVIEW

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java [hunk 1]
```diff
@@ -139,6 +152,55 @@ static DecoratorInfo createDecorator(ClassInfo decoratorClass, BeanDeployment be
                 decoratedTypes, injections, priority);
     }
 
+    /**
+     * Checks if a method belongs to any of the decorated types.
+     * A method belongs to a decorated type if it has the same name and parameter types.
+     */
+    private static boolean belongsToDecoratedType(MethodInfo method, Set<Type> decoratedTypes,
```
Comment (Ladicek) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java]: This method is incorrect, because it ignores methods inherited from superinterfaces of decorated types.

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java [hunk 2]
```diff
@@ -139,6 +152,55 @@ static DecoratorInfo createDecorator(ClassInfo decoratorClass, BeanDeployment be
...
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Checks if two methods have matching parameter types.
+     */
+    private static boolean methodParametersMatch(MethodInfo method1, MethodInfo method2) {
```
Comment (Ladicek) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java]: This method is incorrect, because it ignores generic interfaces and decorators implementing instantiations of them.

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java (hunk 1 above)
Comment (Ladicek) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java]: This should be relatively easy to fix.

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java (hunk 2 above)
Comment (Ladicek) [reply independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Decorators.java]: This can actually be fairly hard to fix. I know we have utilities in ArC to do type parameter substitutions, but I'm not sure how easily they apply here.

**Comment by Ladicek** [comment]
Also, this is completely lacking tests.
